src/server/pfs/server/url.go

```diff
@@ -117,17 +132,24 @@ func putFileURLRecursive(ctx context.Context, taskService task.Service, uw *file
 			}
 			return nil
 		}
-		if err := objClient.Walk(ctx, url.Object, func(p string) error {
+		list := bucket.List(&blob.ListOptions{Prefix: url.Object})
+		var listObj *blob.ListObject
+		for {
+			listObj, err = list.Next(ctx)
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				return errors.Wrapf(err, "error iterating through list for bucket %s", url.Bucket)
+			}
+			log.Debug(ctx, "object stats", zap.String("object", listObj.Key), zap.Int64("size", listObj.Size))
 			if len(paths) >= defaultURLTaskSize {
 				if err := createTask(); err != nil {
-					return err
+					return errors.Wrapf(err, "error from callback on key %s in bucket %s", listObj.Key, url.Bucket)
```

Comment:
I think this message could generally be misleading since the key / bucket will likely not have anything to do with an error here. I think we should just return the error as is.

Reply:
Creating a logger with a bucket field for this section of code would probably be a cleaner way to handle this.